funding/manager.go

```diff
@@ -1594,9 +1602,13 @@ func (f *Manager) handleFundingAccept(peer lnpeer.Peer,
 		implicitChannelType := implicitNegotiateCommitmentType(
 			peer.LocalFeatures(), peer.RemoteFeatures(),
 		)
-		negotiatedChannelType, err := negotiateCommitmentType(
+
+		// We pass in false here as the funder since at this point, we
+		// didn't set a chan type ourselves, so falling back to
+		// implicit funding is acceptable.
+		_, negotiatedChannelType, err := negotiateCommitmentType(
```

Comment:
Will there be the case that we do support explicit channel type but choose not to set a `ChannelType` when opening the channel? Yet the remote node also has the explicit channel type feature bit set, then we would return an error `errUnsupportedChannelType` here?

Reply:
> Will there be the case that we do support explicit channel type but choose not to set a ChannelType when opening the channel?

Yes, that's the case currently which allows us to fall back to implicit if the user doesn't set a type manually. 

> Yet the remote node also has the explicit channel type feature bit set, then we would return an error errUnsupportedChannelType here?

In the latest diff, we pass `false` here, so we won't reach that `errUnsupportedChannelType` case. We enter this clause when: 
  * we don't send a chan type feature bit vector
  * they set one
  * we just wat to make sure what they sent is compatible w/ what we'd arrive at w/ the implicit negotiation

Reply:
The current logic will also let us simplify the itests a bit w.r.t build tags, since we can just specify _any_ type rather than needing to use build tags to set/unset some feature bits.

Reply:
We'll hit the implicit case due to `false` being passed in